articles/tests/test_views.py

```diff
@@ -260,3 +263,186 @@ def test_get_article_with_invalid_slug_fail(self) -> None:
             **self.bearer_token,
         )
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_like_article_pass(self) -> None:
+        """test the ability to like an article"""
+        likes = self.article.likes.count()
+        response = self.client.patch(
+            reverse("article-favorite", kwargs={"slug": self.article.slug}),
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data.get("likes")), likes + 1)
+
+    def test_unlike_article_pass(self) -> None:
+        """test the ability to unlike an article that is already liked. Removes user from the likes list"""
+        likes = self.article.likes.count()
+        response = self.client.patch(
+            reverse("article-favorite", kwargs={"slug": self.article.slug}),
+            **self.bearer_token,
+        )
+        response = self.client.patch(
+            reverse("article-favorite", kwargs={"slug": self.article.slug}),
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data.get("likes")), likes)
+
+    def test_like_unfavorited_article_removes_the_dislike_entry_pass(
+        self,
+    ) -> None:
+        """
+        test that a like on an unfavorited article removes the dislike entry
+        If a user has already disliked an article, and then likes it, the dislike entry should be removed
+        before being added to the like entry
+        """
+        dislikes = self.article.dislikes.count()
+        likes = self.article.likes.count()
+        response = self.client.patch(
+            reverse("article-unfavorite", kwargs={"slug": self.article.slug}),
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data.get("dislikes")), dislikes + 1)
+        response = self.client.patch(
+            reverse("article-favorite", kwargs={"slug": self.article.slug}),
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data.get("likes")), likes + 1)
+        self.assertEqual(len(response.data.get("dislikes")), dislikes)
```

Comment:
Even here shouldn't the dislikes reduce by 1?

Reply:
The logic is correct. Just check through it.